test_unstructured_ingest/evaluation-metrics.sh

```diff
@@ -14,31 +14,54 @@ EVAL_NAME="$1"
 # Download cct test from s3
 BUCKET_NAME=utic-dev-tech-fixtures
 FOLDER_NAME=small-eval-"$EVAL_NAME"
-LOCAL_EVAL_SOURCE_DIR=$SCRIPT_DIR/gold-standard/$FOLDER_NAME
-mkdir -p "$LOCAL_EVAL_SOURCE_DIR"
-aws s3 cp "s3://$BUCKET_NAME/$FOLDER_NAME" "$LOCAL_EVAL_SOURCE_DIR" --recursive --no-sign-request --region us-east-2
+SOURCE_DIR=$SCRIPT_DIR/gold-standard/$FOLDER_NAME
+mkdir -p "$SOURCE_DIR"
+aws s3 cp "s3://$BUCKET_NAME/$FOLDER_NAME" "$SOURCE_DIR" --recursive --no-sign-request --region us-east-2
 
 EXPORT_DIR="$SCRIPT_DIR"/metrics
 
 # shellcheck disable=SC1091
 source "$SCRIPT_DIR"/cleanup.sh
 function cleanup() {
   cleanup_dir "$OUTPUT_DIR"
-  cleanup_dir "$LOCAL_EVAL_SOURCE_DIR"
+  cleanup_dir "$SOURCE_DIR"
 }
 trap cleanup EXIT
 
+# build args
+function generate_args() {
+  local argtype="$1"
+  local dirpath="$2"
+  local list=("${@:3}")
+
+  local -a args
+
+  args=("--${argtype}_dir" "$dirpath")
+  for filename in "${list[@]}"; do
+      args+=("--${argtype}_list" "$filename")
+  done
+  echo "${args[@]}"
+}
+
+# List selected output as a subset of OUTPUT_DIR, if any
+OUTPUT_LIST=(
+)
+# List selected source as a subset of SOURCE_DIR, if any
+SOURCE_LIST=(
+)
+
 if [ "$EVAL_NAME" == "text-extraction" ]; then
-  STRATEGY="measure-text-edit-distance"
+  METRIC_STRATEGY="measure-text-edit-distance"
 elif [ "$EVAL_NAME" == "element-type" ]; then
-  STRATEGY="measure-element-type-accuracy"
+  METRIC_STRATEGY="measure-element-type-accuracy"
 else
-  echo "Wrong evaluation strategy given. Got [ $EVAL_NAME ]."
+  echo "Wrong metric evaluation strategy given. Expected one of [ text-extraction, element-type ]. Got [ $EVAL_NAME ]."
```

Comment:
can we have this input check occur before the small-eval-"$EVAL_NAME" folder gets created? that way a folder with an invalid name (like "small-eval-" on an accidental no EVAL_NAME input) doesn't get created and therefore doesn't need to be cleaned up at the end before exiting.

Reply:
Good catch! Thanks